pkg/registry/file/vulnsummarystorage_test.go

```diff
@@ -0,0 +1,461 @@
+package file
+
+import (
+	"context"
+	"testing"
+
+	"github.com/kubescape/storage/pkg/apis/softwarecomposition"
+	"github.com/kubescape/storage/pkg/apis/softwarecomposition/v1beta1"
+	"github.com/spf13/afero"
+	"github.com/stretchr/testify/assert"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apiserver/pkg/storage"
+)
+
+func TestVulnSummaryStorageImpl_Count(t *testing.T) {
+	files := []string{
+		"/other/type/ns/titi.json",
+		"/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3filtereds/kubescape/titi.json",
+		"/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3filtereds/other/toto.json",
+		"/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/kubescape/toto.json",
+		"/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/other/toto.json",
+	}
+	tests := []struct {
+		name    string
+		key     string
+		want    int64
+		wantErr bool
+	}{
+		{
+			name: "one object",
+			key:  "/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/kubescape/toto",
+			want: 1,
+		},
+		{
+			name: "one ns",
+			key:  "/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s/kubescape",
+			want: 1,
+		},
+		{
+			name: "one type",
+			key:  "/spdx.softwarecomposition.kubescape.io/sbomspdxv2p3s",
+			want: 2,
+		},
+		{
+			name: "all types",
+			key:  "/spdx.softwarecomposition.kubescape.io",
+			want: 4,
+		},
+		{
+			name: "from top",
+			key:  "/",
+			want: 5,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			fs := afero.NewMemMapFs()
+			_ = fs.Mkdir(DefaultStorageRoot, 0755)
+			for _, f := range files {
+				_ = afero.WriteFile(fs, DefaultStorageRoot+f, []byte(""), 0644)
+			}
+			s := NewStorageImpl(fs, DefaultStorageRoot)
```

Comment:
issue: this tests the `Count()` test for SBOMs and has no relation to the summary objects

We have no need for this, so it should be adapted to the actual implementation of the `Count()` object of this storage.